crates/codegen/src/compiler/functions.rs

```diff
@@ -932,11 +938,19 @@ impl FunctionCompiler<'_> {
                     return None;
                 }
 
-                let (_, sub_ty) = ty.as_array().expect("array literals must have array types");
+                let array_size = if let Some(sub_ty) = ty.as_slice() {
```

Comment:
Since the syntax changes, slice literals have been removed (`[]i32{..}` vs `[3]i32{..}`) so you could just use `ty.size()` here instead of checking if the array literal has a slice type